modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -499,12 +517,67 @@ CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
       addMeshSubgenerator("PlaneIDMeshGenerator", build_mesh_name, params);
     }
   }
-  else
-    declareMeshProperty("extruded", false);
+  generateMetadata();
 
   _build_mesh = &getMeshByName(build_mesh_name);
 }
 
+void
+CoreMeshGenerator::generateMetadata()
+{
+  // Define metadata related to downstream function calls
+  if (_mesh_periphery)
+  {
+    declareMeshProperty(RGMB::peripheral_ring_radius, _outer_circle_radius);
+    declareMeshProperty(RGMB::peripheral_ring_region_id, _periphery_region_id);
+  }
+
+  // Determine constituent pin type ids and define lattice
+  std::vector<std::vector<int>> assembly_name_lattice;
+  std::vector<std::string> input_assembly_names;
+  std::vector<std::string> input_pin_names;
+  for (unsigned int i = 0; i < _pattern.size(); ++i)
+  {
+    std::vector<int> assembly_name_idx(_pattern[i].size());
+    for (unsigned int j = 0; j < _pattern[i].size(); ++j)
+    {
+      const auto input_assembly_name = _inputs[_pattern[i][j]];
+      // Use an assembly type of -1 to represent a dummy assembly
+      if (input_assembly_name == _empty_key)
+        assembly_name_idx[j] = -1;
+      else
+      {
+        const auto it = std::find(
+            input_assembly_names.begin(), input_assembly_names.end(), input_assembly_name);
+        if (it == input_assembly_names.end())
+        {
+          assembly_name_idx[j] = input_assembly_names.size();
+          input_assembly_names.push_back(input_assembly_name);
+          if (!getMeshProperty<bool>(RGMB::is_single_pin, input_assembly_name))
+          {
+            const auto pin_names =
+                getMeshProperty<std::vector<std::string>>(RGMB::pin_names, input_assembly_name);
+            for (const auto & pin_name : pin_names)
+              if (std::find(input_pin_names.begin(), input_pin_names.end(), pin_name) ==
+                  input_pin_names.end())
+                input_pin_names.push_back(pin_name);
+          }
+        }
+        else
+          assembly_name_idx[j] = it - input_assembly_names.begin();
```

Comment:
why overwrite the value with the last one you see?

Reply:
I don't understand what you mean. This line basically populates `assembly_name_idx` with the index of where `assembly_name` occurs in `input_assembly_names` (`it` has value of `std::find(...)`)

Reply:
oh yeah I had the logic flipped